### ISSUE TITLE
[BUG] Accessing `$rootSpan->getService()` of internal span fails with exception

### ISSUE THREAD

**Issue by olsavmic** [DataDog/dd-trace-php#1317]
### Bug description

**ddtrace-php v0.64.1**

Accessing `$rootSpan->getService()` of internal span fails with `Exception thrown in ddtrace's closure for ShipMonk\Console\Application::doRunCommand(): Undefined property: DDTrace\SpanData::$service`.

The problem disappears when setting `DD_TRACE_GENERATE_ROOT_SPAN=true`.

```php
trace_method(
    Application::class,
    'doRunCommand',
    function (SpanData $spanData, $args, $retval, ?Throwable $throwable): void {
        $tracer = GlobalTracer::get();
        $rootSpan = $tracer->getSafeRootSpan();
        if ($rootSpan === null) {
            return;
        }

        $rootSpan->getService();
});
```

**ENV**
```
DD_TRACE_GENERATE_ROOT_SPAN=false
```

### PHP version

PHP 8.0.10 (cli) (built: Aug 26 2021 16:09:29) ( NTS )

### Installed extensions

[PHP Modules]
bcmath
blackfire
calendar
Core
ctype
curl
date
ddtrace
dom
exif
FFI
fileinfo
filter
ftp
gettext
hash
iconv
igbinary
imagick
imap
intl
json
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
soap
sockets
sodium
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
blackfire
ddtrace

### OS info

NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster

**Comment by bwoebi** [comment]
That's true, the object properties are not set. There's nothing meaningful, unless you set a service name yourself. So e.g. if the first line of your trace method would be `$spanData->service = "foo";`, the getService() would return `foo` too.

I'm not sure whether that's actually a bug.

**Comment by olsavmic** [comment]
Root span by default takes the value from DD_SERVICE env so I'd expect that'd be the case.

But even if it's by design, it's unexpected behaviour as the getService function definition specifies that it returns string, not string|null or `@throws`.

**Comment by bwoebi** [comment]
Will be fixed with #1329 (i.e. in next release).
